SalesforceCommerceCloud/pwa-kit
# @W-16571700 Alter analytics behavior for einstein based on DNT

# Description

Apply the DNT hook in the Einstein layer of template retail react app. This will also be an example of how to integrate DNT into other analytics layers. 
# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

- (step1) Run the E2E test for dnt.spec.js and see that it passes
^ To do this, 
1. Go to the root of the pwa-kit repository.
2. Run `npm run test:e2e:ui`
3. Run dnt.spec.js
- (step2) Go to https://runtime-admin-n16.mobify-storefront.com/global/en-GB/category/womens. Don't answer the consent tracking form. Open network tab and filter for "einstein". 
- (step3) Click on different products to see that 'viewed-recently-einstein' network requests are being sent
- (step3) Answer "Decline" on the consent tracking form, and now go to different products to see that those einstein requests are no longer being sent

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
